packages/react/src/components/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,379 @@
+import classNames from '../../utils/classNames';
+import { IconAngleDown, IconCheck, IconAttention } from '../../icons';
+import Checkbox from '../checkbox/Checkbox';
+
+type OptionType = {
+  [key: string]: any;
+};
+
+export type DropdownProps = {
+  /**
+   * Controls the circular keyboard navigation between items. If set to `true`, when first item is highlighted, the Arrow Up will move highlight to the last item, and vice versa using Arrow Down.
+   */
+  circularNavigation?: boolean;
+  /**
+   * Additional class names to apply to the dropdown
+   */
+  className?: string;
+  /**
+   * Close the menu when the user selects an option
+   */
+  closeMenuOnSelect?: boolean;
+  /**
+   * If `true`, the dropdown will be disabled
+   */
+  disabled?: boolean;
+  /**
+   * If `true`, the dropdown menu can be filtered
+   */
+  filterable?: boolean;
+  /**
+   * Function used to get `aria` attributes and the `id` prop for menu options (`li`)
+   * @param index Data item index
+   */
+  getItemId?: (index: number) => string;
+  /**
+   * The helper text content that will be shown below the dropdown
+   */
+  helper?: string | ReactNode;
+  /**
+   * Hides the label above the dropdown
+   */
+  hideLabel?: boolean;
+  /**
+   * Item that should be selected when the dropdown is initialized
+   */
+  defaultValue?: OptionType;
+  /**
+   * Item(s) that should be selected when the dropdown is initialized. Use this instead of `initialSelectedItem` when `multiselect` is enabled
+   */
+  defaultValues?: OptionType[];
+  /**
+   * Used to generate the first part of the id on the elements.
+   * You can override this id with one of your own, provided as a prop, or you can override the id for each element using the `getItemId`, `labelId`, `inputId`, `menuId` and `toggleButtonId` props.
+   */
+  id?: string;
+  /**
+   * If `true`, the input and `helper` will be displayed in an invalid state
+   */
+  invalid?: boolean;
+  /**
+   * A function used to detect whether an option is disabled ([example](/?path=/story/components-dropdown--disabled-options))
+   * @param option  Data item
+   * @param index   Data item index
+   */
+  isOptionDisabled?: (option: OptionType, index: number) => boolean;
+  /**
+   * Sets the data item field that represents the item label.
+   * E.g. an `optionLabelField` value of `'foo'` and a data item `{ foo: 'Label', bar: 'value' }`, would display `Label` in the menu for that specific item
+   */
+  optionLabelField?: string;
+  /**
+   * Used for `aria` attributes and the `id` prop for the label element
+   */
+  labelId?: string;
+  /**
+   * The label for the dropdown
+   */
+  label?: string | ReactNode;
+  /**
+   * Used for `aria` attributes and the `id` prop for the input element when `filterable` is `true`
+   */
+  inputId?: string;
+  /**
+   * Used for `aria` attributes and the `id` prop for the menu (`ul`)
+   */
+  menuId?: string;
+  /**
+   * Enables selecting multiple values if `true`
+   */
+  multiselect?: boolean;
```

Comment:
Should contain a comment about `multiselect` not working in tandem with `filterable`..?